README.md

```diff
@@ -1,13 +1,161 @@
+
+## Contents
+1. [Get started](#get-started)
+2. [Platform dependencies](#platform-dependencies)
+3. [XML doc generation](#xml-doc-generation)
+4. [Installation](#installation)
+5. [Documentation](#documentation)
+6. [Features](#features)
+    * [Authorization](#authorize)
+    * [Upload a document to SignNow](#upload-document)
+    * [Download a document from SignNow](#download-document)
+    * [Create a single-use link to the document for signature](#create-signing-link)
+7. [Important notes](#important-notes)
+8. [License](#license)
+
+
+## <a name="get-started"></a>Get started
+  * Get a SignNow account [here](https://www.signnow.com/developers) 
+  * SignNow uses Authorization: Basic <credentials>. To get your access token, have your client ID and secret base-64 encoded first
+
+
+## <a name="platform-dependencies"></a>Platform Dependencies
+#### Windows
+  * .Net Framework 4.5 or newer version should be installed in your system, or
+  * .Net Standard 1.2 or 2.0, or
+  * .Net Core 2.2 and newer
+
+#### MacOS and Linux
+  * .Net Core 2.2 and newer
+
+
+## <a name="xml-doc-generation"></a>XML doc generation
+
+For XML documentation generation, install InheritDocTool (the project build will fail without it):
+
+```bash
 dotnet tool install -g InheritDocTool
 ```
 
-You can see details about InheritDoc [here](https://www.inheritdoc.io)
+More about the InheritDoc [here](https://www.inheritdoc.io)
+
+
+## <a name="installation"></a>Installation
+
+### Install .Net package
+
+```bash
+dotnet add package SignNow.Net --version <package-version>
+```
+
+### Install using Package manager
+
+```powershell
+Install-Package SignNow.Net -Version <package-version>
+```
+
+## <a name="documentation"></a>Documentation
+
+Read about the available SignNow features in [SignNow API Docs](https://github.com/signnow/SignNow.NET.wiki.git) .
+
+
+## <a name="features"></a>Features
+
+### <a name="authorization"></a> Authorization
+
+Get your access token via OAuth 2.0 service.
+
+```csharp
+string clientId = "0fa****-EXAMPLE_CLIENT_ID-****13";
+string clientSecret = "0fb**-EXAMPLE_CLIENT_SECRET-**13";
+
+string userLogin = "signnow_dotnet_sdk@example.com";
+string userPassword = "example-user-password";
+
+var oauth = new OAuth2Service(clientId, clientSecret);
+
+var token = oauth.GetTokenAsync(userLogin, userPassword, Scope.All).Result;
+}
+```
+
+### <a name="upload-document"></a> Upload a document to SignNow
+
+All the features in SignNow require a `document_id`. Once you upload a document to SignNow, you get the `document_id` from a successful response.
+```csharp
+string documentId = default;
+
+var pdfFilePath = "./file-path/document-example.pdf";
+var pdfFileName = "document-example.pdf";
+
+/// using token from the Authorization step
+var documentService = new DocumentService(token);
+
+using (var fileStream = File.OpenRead(pdfFilePath))
+{
+    var uploadResponse = documentService.UploadDocumentAsync(fileStream, pdfFileName, default).Result;
```

Comment:
why "default" in the last param?

Reply:
explicit interface usage. Removed optional param to simplify example